index.js

```diff
@@ -109,6 +169,266 @@ export const immediate = x => Future ((rej, res) => {
+//. 2. log an error to the console if:
+//.     * A network error occurs.
+//.     * The reponse code is not 200.
+//.     * The JSON is malformed.
+//.
+//. Note that we were in control of how an unexpected status was treated,
+//. how an erroneous response would be formatted as an error message,
+//. whether the response would be parsed as JSON, and how a failure of parsing
+//. the JSON would have been treated.
+//.
+//. The goal of the functions below us to give you as much control over HTTP
+//. requests as possible, while still keeping boilerplate low by leveraging
+//. function composition.
+//.
+//. This contrasts with many of the popular HTTP client libraries out there,
+//. which often make decisions for you, taking away control in an attempt to
+//. provide a smoother usage experience.
+
+// getAgent :: String -> Future Error Agent
+const getAgent = protocol => {
+  switch (protocol) {
+  case 'https:': return resolve (https);
+  case 'http:': return resolve (http);
+  default: return reject (new Error (`Unsupported protocol '${protocol}'`));
+  }
+};
+
+//# request :: Object -> Url -> Readable Buffer -> Future Error IncomingMessage
+//.
+//. Given an Object of [http options][], a String containing the request URL,
+//. and a [Readable][] stream of [Buffer][]s to be sent as the request body,
+//. returns a Future which makes an HTTP request and resolves with its
+//. [IncomingMessage][]. If the Future is cancelled, the request is aborted.
+//.
+//. This is the "lowest level" function for making HTTP requests. It does not
+//. handle buffering, encoding, content negotiation, or anything really.
+//. For most use-cases, you can use one of the more specialized functions:
+//.
+//. * [`send`](#sent): Make a generic HTTP request.
+//. * [`retrieve`](#retrieve): Make a GET request.
+export const request = options => url => body => {
+  const location = new URL (url);
+  return chain (agent => Future ((rej, res) => {
+    const req = agent.request (location, options, res);
+    pipeline (body, req, e => e && rej (e));
+    return () => {
+      req.abort ();
+    };
+  })) (getAgent (location.protocol));
+};
+
+//# messageMeta :: IncomingMessage -> MessageMeta
+//.
+//. Extracts the common meta-data from an [IncomingMessage][].
+//.
+//. The `MessageMeta` type is defined as
+//. `{ code :: Number, message :: String, headers :: StrMap String }`:
+//.
+//. * `code`: The HTTP status code.
+//. * `message`: The HTTP status message.
+//. * `headers`: A StrMap of the HTTP headers.
+export const messageMeta = message => ({
+  code: message.statusCode,
+  message: message.statusMessage,
+  headers: message.headers,
+});
+
+//# bufferResponse :: Charset -> IncomingMessage -> Future Error {meta :: MessageMeta, body :: String}
+//.
+//. A version of [`buffer`](#buffer) specialized in [IncomingMessage][]s.
+//.
+//. Given a charset and an IncomingMessage, returns a Future with the
+//. [MessageMeta](#messagemeta) and the buffered, encoded, message body.
```

Comment:
Is this meant to link to `MessageMeta` or to `messageMeta`?

Reply:
It links to the `messageMeta` function, where users will be able to find what the `MessageMeta` type is.

Reply:
Why not use `#messageMeta` as the fragment identifier? :)

Reply:
I don't think that's supported on GitHub, right? Unless I create the anchor tag myself.

Reply:
Will you use Transcribe to generate the readme? If so, you will have a second anchor with the correctly capitalized identifier. :)

Reply:
Oh. Right! :)